src/pybind/mgr/diskprediction/module.py

```diff
@@ -0,0 +1,346 @@
+            else:
+                msg = json.dumps(osd_data, indent=4)
+        except Exception as e:
+            if str(e).find('No such file') >= 0:
+                msg = 'unable to get osd %s predicted data'
+            else:
+                msg = 'unable to get osd %s predicted data, %s' % (cmd['osd_id'], str(e))
+            self.log.error(msg)
+        return 0, msg, ''
+
+    def _config_set(self, inbuf, cmd):
+        self.set_config('diskprediction_server', cmd['server'])
+        self.set_config('diskprediction_user', cmd['user'])
+        self.set_config('diskprediction_password', cmd['password'])
+        if cmd.get('port'):
+            self.set_config('diskprediction_port', cmd['port'])
+        self.show_module_config()
+        return 0, json.dumps(self.config, indent=4), ''
+
+    def _run_prediction_forced(self, inbuf, cmd):
+        msg = ''
+        for _task in self._tasks:
+            if isinstance(_task, PredictionTask):
+                msg = 'run prediction agent successfully'
+                _task.event.set()
+        return 0, msg, ''
+
+    def _run_metrics_forced(self, inbuf, cmd):
+        msg = ''
+        for _task in self._tasks:
+            if isinstance(_task, MetricsTask):
+                msg = 'run metrics agent successfully'
+                _task.event.set()
+        return 0, msg, ''
+
+    def _run_smart_forced(self, inbuf, cmd):
+        msg = ' '
+        for _task in self._tasks:
+            if isinstance(_task, SmartTask):
+                msg = 'run smart agent successfully'
+                _task.event.set()
+        return 0, msg, ''
+
+    def _status(self, inbuf, cmd):
+        msg = 'diskprediction plugin status: %s' % self.status
+        return 0, msg, ''
+
+    def handle_command(self, inbuf, cmd):
+        for o_cmd in self.COMMANDS:
+            if cmd['prefix'] == o_cmd['cmd'][:len(cmd['prefix'])]:
+                fun = getattr(
+                    self, '_%s' % o_cmd['cmd'].split(' ')[1].replace('-', '_'))
+                if fun:
+                    return fun(inbuf, cmd)
+        return 0, 'cmd not found', ''
+
+    def show_module_config(self):
+        self.fsid = self.get('mon_map')['fsid']
+        self.log.debug('Found Ceph fsid %s', self.fsid)
+
+        for key, default in self.config_keys.items():
+            self.set_config_option(key, self.get_config(key, default))
+
+    def serve(self):
+        self.log.info('Starting diskprediction module')
+        self.run = True
+        self.status = DP_MGR_STAT_ENABLED
+
+        while self.run:
+            if self.get_configuration("diskprediction_config_mode").lower() in ['cloud', 'onpremise']:
+                enable_colud = True
+            if enable_colud and not self._activate_cloud:
+                self.start_cloud_disk_prediction()
+            elif not enable_colud and self._activate_cloud:
+                self.stop_cloud_disk_prediction()
+            self.event.wait(60)
```

Comment:
Since all we're doing is waking up and checking a configuration value, this could be something like 5 seconds instead of 60.  Otherwise when someone enables it they're going to be waiting a minute and wondering why nothing happens.